Utils/src/CandidateTrackFilter.cc

```diff
@@ -67,6 +70,21 @@ class TrackInfos {
 		pfcands_pvassociationquality = std::make_unique<vector<int>>();
 		pfcands_dzassociatedpv       = std::make_unique<vector<double>>();
 		pfcands_vtxidx               = std::make_unique<vector<int>>();
+		vtx_sumtrackpt2              = std::make_unique<vector<double>>();
+	}
+
+	void calculateVertexQuantities(const int & nvtx) {
```

Comment:
wouldn't it be more efficient to use the stored vertex indices that are already computed for the tracks? just loop over all tracks, adding the track pt2 to the vector entry corresponding to the vertex index

Reply:
Okay, I changed this function to use the existing indices. I checked and it's functionally equivalent.

Reply:
primitive types should be passed by copy, rather than const ref

Reply:
Changed. Out of curiosity, why is that? I know it won't make functionally change anything, but is there a penalty in time or memory to passing a const ref?

Reply:
Yes, it's actually faster to copy a primitive type than to pass a reference. For a complex class, it's faster to pass a reference.